adaptors/dell-hwmgr/hwmgrclient/client.go

```diff
@@ -336,3 +336,44 @@ func (c *HardwareManagerClient) GetSecret(ctx context.Context, secretKey string)
 
 	return response.JSON200, nil
 }
+
+// ValidateResourceGroup validates the hardware manager resource group data with nodepool
+func (c *HardwareManagerClient) ValidateResourceGroup(
+	ctx context.Context,
+	nodepool *hwmgmtv1alpha1.NodePool,
+	resourceGroup hwmgrapi.RhprotoResourceGroupObjectGetResponseBody,
+) error {
+	if *resourceGroup.ResourceSelectors != nil {
+		resourceSelector := *resourceGroup.ResourceSelectors
+		for _, node := range nodepool.Spec.NodeGroup {
```

Comment:
The iterator here isn't a node, it's a nodegroup.
```suggestion
		for _, nodegroup := range nodepool.Spec.NodeGroup {
```